src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -358,4 +402,8 @@ private static <T> org.jooq.Field<T> valueField(T value, FieldCondition<?> condi
     }
     return val(value);
   }
+  public static boolean isValidUUID(String uuidStr){
+    UUID uuid = UUID.fromString(uuidStr);
+    return uuid.toString().equals(uuidStr);
+  }
```

Comment:
Does this work for checking whether something is a UUID? Seems like it might just throw an IllegalArgumentException if it doesn't match.

Reply:
You could make the method void and do a try-catch block that catches IllegalArgumentException, and throws InvalidFqlException from the the catch block. Then you could call that method from the outer methods.